src/Arachne.cc

```diff
@@ -658,7 +658,7 @@ dispatch() {
         // At this point, it is guaranteed that it is safe to read the context
         // information.
         ThreadContext* currentContext = core.localThreadContexts[currentIndex];
-        if (currentIndex == core.highestOccupiedContext + 1) {
+        if (currentIndex > core.highestOccupiedContext) {
```

Comment:
@jiazhang0, have you measured a difference in performance here? I believe the original code makes the intent more clear, so I'd only change it if it made a measurable difference.